pkg/operator/controller/dns/controller.go

```diff
@@ -251,17 +251,34 @@ func (r *reconciler) publishRecordToZones(zones []configv1.DNSZone, record *iov1
 		// Only publish the record if the DNSRecord has been modified
 		// (which would mean the target could have changed) or its
 		// status does not indicate that it has already been published.
-		if record.Generation == record.Status.ObservedGeneration && recordIsAlreadyPublishedToZone(record, &zone) {
-			log.Info("skipping zone to which the DNS record is already published", "record", record.Spec, "dnszone", zone)
-			continue
-		}
 
 		condition := iov1.DNSZoneCondition{
 			Status:             string(operatorv1.ConditionUnknown),
 			Type:               iov1.DNSRecordFailedConditionType,
 			LastTransitionTime: metav1.Now(),
 		}
 
+		if record.Generation != record.Status.ObservedGeneration && recordIsAlreadyPublishedToZone(record, &zone) {
+			log.Info("replacing DNS record", "record", record.Spec, "dnszone", zone)
+
+			if err := r.dnsProvider.Replace(record, zone); err != nil {
+				log.Error(err, "failed to replace DNS record to zone", "record", record.Spec, "dnszone", zone)
+				condition.Status = string(operatorv1.ConditionTrue)
+				condition.Reason = "ProviderError"
+				condition.Message = fmt.Sprintf("The DNS provider failed to replace the record: %v", err)
+				result.RequeueAfter = 30 * time.Second
+			} else {
+				log.Info("replaced DNS record to zone", "record", record.Spec, "dnszone", zone)
+				condition.Status = string(operatorv1.ConditionFalse)
+				condition.Reason = "ProviderSuccess"
+				condition.Message = "The DNS provider succeeded in replacing the record"
+			}
+
```

Comment:
You need to update `statuses` and then `continue` here to avoid executing the `Ensure` logic below (and failing).  Alternatively, add an `else` block and pull the `Ensure` logic into it, which would be simpler and more readable:
```go
		if record.Generation == record.Status.ObservedGeneration && recordIsAlreadyPublishedToZone(record, &zone) {
			log.Info("skipping zone to which the DNS record is already published", "record", record.Spec, "dnszone", zone)
			continue
		}

		condition := iov1.DNSZoneCondition{
			Status:             string(operatorv1.ConditionUnknown),
			Type:               iov1.DNSRecordFailedConditionType,
			LastTransitionTime: metav1.Now(),
		}
		if recordIsAlreadyPublishedToZone(record, &zone) {
			if err := r.dnsProvider.Replace(record, zone); err != nil {
				log.Error(err, "failed to replace DNS record", "record", record.Spec, "dnszone", zone)
				condition.Status = string(operatorv1.ConditionTrue)
				condition.Reason = "ProviderError"
				condition.Message = fmt.Sprintf("The DNS provider failed to replace the record: %v", err)
				result.RequeueAfter = 30 * time.Second
			} else {
				log.Info("replaced DNS record", "record", record.Spec, "dnszone", zone)
				condition.Status = string(operatorv1.ConditionFalse)
				condition.Reason = "ProviderSuccess"
				condition.Message = "The DNS provider succeeded in replacing the record"
			}
		} else {
			if err := r.dnsProvider.Ensure(record, zone); err != nil {
				log.Error(err, "failed to publish DNS record to zone", "record", record.Spec, "dnszone", zone)
				condition.Status = string(operatorv1.ConditionTrue)
				condition.Reason = "ProviderError"
				condition.Message = fmt.Sprintf("The DNS provider failed to ensure the record: %v", err)
				result.RequeueAfter = 30 * time.Second
			} else {
				log.Info("published DNS record to zone", "record", record.Spec, "dnszone", zone)
				condition.Status = string(operatorv1.ConditionFalse)
				condition.Reason = "ProviderSuccess"
				condition.Message = "The DNS provider succeeded in ensuring the record"
			}
		}
		statuses = append(statuses, iov1.DNSZoneStatus{
			DNSZone:    zone,
			Conditions: []iov1.DNSZoneCondition{condition},
		})
```